src/Network/Network/ApplicationGateway/Listener/AzureApplicationGatewayListenerBase.cs

```diff
@@ -81,6 +81,11 @@ public class AzureApplicationGatewayListenerBase : NetworkBaseCmdlet
         [ValidateNotNullOrEmpty]
         public string Protocol { get; set; }
 
+        [Parameter(
+            HelpMessage = "Host names")]
+        [ValidateNotNullOrEmpty]
+        public string[] HostNames { get; set; }
```

Comment:
HostName

Reply:
The parameter HostNames in ApplicationGatewayListener has been used to maintain parity with the parameter
 Hostnames in ApplicationGatewayHttpListener.
 https://learn.microsoft.com/en-us/powershell/module/az.network/add-azapplicationgatewayhttplistener?view=azps-10.4.1
 All the downstream NRP code for this has  been implemented following the same motivation 
 and draws inspiration from existing naming conventions.